src/core/expr/fexpr_count_countna.cc

```diff
@@ -0,0 +1,165 @@
+      if (arg_->get_expr_kind() != Kind::None) out += arg_->repr();      
+      out += ')';
+      return out;
+    }
+
+
+    Workframe evaluate_n(EvalContext &ctx) const override {
+      Workframe outputs(ctx);
+      Workframe wf = arg_->evaluate_n(ctx);
+      Groupby gby = ctx.get_groupby();
+      // this covers scenarios where
+      // we dont care about the presence or absence of NAs
+      // we just want the total number of rows
+      bool count_all_rows = arg_->get_expr_kind() == Kind::None;
+
+      if (count_all_rows && !COUNTNA) {
+        Column coli;
+        if (gby){
+          coli = Column(new Latent_ColumnImpl(
+                new CountAllRows_ColumnImpl(gby)
+              ));
+        } else {
+            auto value = static_cast<int64_t>(ctx.nrows());
+            coli = Const_ColumnImpl::make_int_column(1, value, SType::INT64);
+          }
+        outputs.add_column(std::move(coli), "count", Grouping::GtoONE);
+        return outputs;
+      }
+
+      if (!gby) {
+        gby = Groupby::single_group(wf.nrows());
+      } 
+
+      for (size_t i = 0; i < wf.ncols(); ++i) {
+        bool is_grouped = ctx.has_group_column(
+                            wf.get_frame_id(i),
+                            wf.get_column_id(i)
+                          );
+        Column coli = wf.retrieve_column(i);   
+        if (COUNTNA && !ctx.has_groupby() && (coli.stype() == SType::VOID)) {
+          int64_t nrows = static_cast<int64_t>(ctx.nrows());
+          coli = Const_ColumnImpl::make_int_column(1, nrows, SType::INT64);
+        } else {coli = evaluate1(std::move(coli), gby, is_grouped);}
+        outputs.add_column(std::move(coli), wf.retrieve_name(i), Grouping::GtoONE);                         
+      }        
+      return outputs;
+    }
+
+
+    Column evaluate1(Column &&col, const Groupby& gby, bool is_grouped) const {
+      SType stype = col.stype();
+      switch (stype) {
+        case SType::VOID:
+        case SType::BOOL:
+        case SType::INT8:
+          return make<int8_t, int64_t>(std::move(col), gby, is_grouped);
+        case SType::INT16:
+          return make<int16_t, int64_t>(std::move(col), gby, is_grouped);
+        case SType::DATE32: 
+        case SType::INT32:
+          return make<int32_t, int64_t>(std::move(col), gby, is_grouped);
+        case SType::TIME64: 
+        case SType::INT64:
+          return make<int64_t, int64_t>(std::move(col), gby, is_grouped);
+        case SType::FLOAT32:
+          return make<float, int64_t>(std::move(col), gby, is_grouped);
+        case SType::FLOAT64:
+          return make<double, int64_t>(std::move(col), gby, is_grouped);
+        case SType::STR32:
+        case SType::STR64:
+          return make<CString, int64_t>(std::move(col), gby, is_grouped);                
+        default:
+          throw TypeError()
+            << "Invalid column of type `" << stype << "` in " << repr();
+      }
+    }
+
+
+    template <typename T, typename U>
```

Comment:
I'm not sure why we need this additional `U` parameter since it is always `int64_t`. I can imagine we may have cases when the return type depends on the column type, but it is not the case for `count()` and `countna()`.